mayastor/src/bdev/nexus/nexus_share.rs

```diff
@@ -88,57 +85,101 @@ impl Nexus {
         debug!("creating share handle for {}", name);
         // The share handle is the actual bdev that is shared through the
         // various protocols.
-        let disk = Disk::create(&name).await.context(ShareNexus {
-            name: self.name.clone(),
-        })?;
-        let device_path = disk.get_path();
+
+        let return_val = match share_protocol {
+            ShareProtocol::Nbd => {
+                // Publish the nexus to system using nbd device and return the path to
+                // nbd device.
+                let nbd_disk =
+                    NbdDisk::create(&name).await.context(ShareNbdNexus {
+                        name: self.name.clone(),
+                    })?;
+                let device_path = nbd_disk.get_path();
+                self.nbd_disk = Some(nbd_disk);
+                Ok(device_path)
+            },
+            ShareProtocol::Iscsi => {
+                // Publish the nexus to system using an iscsi target and return the IQN
+                let iscsi_target =
+                    NexusIscsiTarget::create(&name).context(ShareIscsiNexus {
+                        name: self.name.clone(),
+                    })?;
+                let iqn = iscsi_target.get_iqn();
+                self.iscsi_target = Some(iscsi_target);
+                Ok(iqn)
+            },
+            ShareProtocol::Nvmf => {
+                Err(Error::InvalidShareProtocol {sp_value: share_protocol as i32})
```

Comment:
This should be return Err....
otherwise code following this will be executed, which I believe is not the intention.

Reply:
Already fixed in commit:
https://github.com/tjoshum/Mayastor/pull/6/commits/50ebfc32484fa9a0eefef47f6c4a1505580980fb